core/shared/src/main/scala/org/typelevel/log4cats/SamStructuredLogger.scala

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats
+
+import cats.*
+import cats.data.{EitherT, Kleisli, OptionT}
+import cats.Show.Shown
+
+/**
+ * A SAM-based StructuredLogger that extends Logger and provides structured logging capabilities.
+ * This implementation uses the new SAM LoggerKernel design for better performance and middleware
+ * compatibility.
+ */
+trait SamStructuredLogger[F[_]] extends Logger[F] {
+  protected def kernel: LoggerKernel[F, String]
+
+  def trace(ctx: Map[String, String])(msg: => String): F[Unit] =
```

Comment:
We should be able to make these final, as they delegate to `kernel`, which should be the overwritten field.